### PR TITLE
Remove user_location.

### PR DESCRIPTION
**Pull request by brianhw**
Pull out util/geolocation utility, since it's shared between location_per_course and events_obfuscation.  

@mulby , @HassanJaveed84

### CODE REVIEW

**Comment by brianhw** [comment]
Acceptance tests all pass on this branch.

File: edx/analytics/tasks/util/geolocation.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import pygeoip
+import tempfile
+
+import luigi
+
+UNKNOWN_COUNTRY = "UNKNOWN"
+UNKNOWN_CODE = "UNKNOWN"
+
+
+class GeolocationMixin(object):
```
Comment (mulby) [inline edx/analytics/tasks/util/geolocation.py]: Maybe rename to `GeolocationDownstreamMixin` and the class below to `GeolocationMixin`?

Generally when I see a class that looks like `FooTask` I assume that it "owns" the `run()` method...

**Review by mulby — APPROVED** [review]
One nit, otherwise LGTM!

File: edx/analytics/tasks/util/geolocation.py (hunk 1 above)
Comment (brianhw) [reply edx/analytics/tasks/util/geolocation.py]: Good idea!

**Comment by brianhw** [comment]
Refactored GeolocationMixin to move all the setup into it.    Ready for review again.

@mulby @HassanJaveed84

**Review by mulby — APPROVED** [review]
LGTM

**Pull request merged**